add forks api

terakilobyte opened the pull request:
Thanks for this library! I'd like to use it for a project I have in mind that involves using the [`forks`](https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#forks) API and thought I'd try adding it to octocrab.

I'm relatively new to Rust so if I've done something silly please don't hesitate to call it out, and if this is an unwelcome addition that's ok too.

On a side note, it appears dependabot updated tokio to 1.0, but reqwest is currently on 0.2. Unless you manually set it back, it results in a panic similar to [this](https://github.com/tokio-rs/tokio/issues/1837) when you run any example or use it in your own project and specify tokio 1.0 :frowning_face:

XAMPPRocky reviewed (changes requested):
Thank you for your PR! Overall this looks good, I've left some comments of some minor changes, and once they've been implemented we can merge this.

> On a side note, it appears dependabot updated tokio to 1.0, but reqwest is currently on 0.2. Unless you manually set it back, it results in a panic similar to this when you run any example or use it in your own project and specify tokio 1.0 ☹️

That's unfortunate, I'll revert that change.

terakilobyte commented on src/api/repos/forks.rs:
````diff
@@ -0,0 +1,126 @@
+    /// let forks = octocrab::instance().repos("owner", "repo").list_forks().sort(ForkSort::Oldest).send().await?;
+    /// # Ok(())
+    /// # }
+    /// ```
````
I'll give it a try.

terakilobyte commented on src/api/repos/forks.rs:
````diff
@@ -0,0 +1,126 @@
+    /// Creates a fork of a repository.
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// let new_fork = octocrab::instance().repos("owner", "repo").create_fork().send().await?;
````
I'll give it a try.

terakilobyte commented:
I've modified the documentation, and added it into the entrance docs in a separate commit for easier removal if it's not wanted. 

One thing I noticed when updating the example and the documentation was that `Into<u8>` seems to infer the type correctly `per_page(30)`, but `Into<u32>` wasn't and I had to explicitly specify the type.

XAMPPRocky commented on src/params.rs:
```diff
@@ -272,6 +272,18 @@ pub mod repos {
+        #[derive(Debug, Clone, Copy, serde::Serialize)]
+        #[serde(rename_all = "snake_case")]
+        #[non_exhaustive]
+
```
```suggestion
```

XAMPPRocky commented on src/api/repos/forks.rs:
````diff
@@ -0,0 +1,114 @@
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// use octocrab::params::repos::forks::Sort;
+    /// let forks = octocrab::instance().repos("owner", "repo").list_forks().sort(Sort::Oldest).page(2u32).per_page(30).send().await?;
````
```suggestion
    /// let forks = octocrab::instance()
    ///     .repos("owner", "repo")
    ///     .list_forks()
    ///     // Optional Parameters
    ///     .sort(Sort::Oldest)
    ///     .page(2u32)
    ///     .per_page(30)
    ///     .send()
    ///     .await?;
```

XAMPPRocky commented on src/api/repos/forks.rs:
````diff
@@ -0,0 +1,114 @@
+    /// create the fork in.
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// let new_fork = octocrab::instance().repos("owner", "repo").create_fork().organization("weyland-yutani").send().await?;
````
```suggestion
    /// let new_fork = octocrab::instance()
    ///     .repos("owner", "repo")
    ///     .create_fork()
    ///     // Optional Parameters
    ///     .organization("weyland-yutani")
    ///     .send()
    ///     .await?;
```

XAMPPRocky commented on src/api/repos/forks.rs:
```diff
@@ -0,0 +1,114 @@
+impl<'octo> RepoHandler<'octo> {
+    /// List forks of a repository. Optionally, specify the
+    /// [sort](./forks/struct.ListForksBuilder.html#method.sort) order,
+    /// [page](./forks/struct.ListForksBuilder.html#method.page),
```
Can you use [intra-doc](https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html) links to link to methods?

terakilobyte replied:
done

XAMPPRocky commented:
>One thing I noticed when updating the example and the documentation was that Into<u8> seems to infer the type correctly per_page(30), but Into<u32> wasn't and I had to explicitly specify the type. I'm not sure why this is. This is a minimal reproduction in the rust playground.

Yeah that is a quirk of the language, specifying with a type suffix (e.g. `1u32`) is fine.

XAMPPRocky commented:
Thank you again for your PR, and congrats on your first contribution! 🎉

XAMPPRocky commented:
I've published this change as part of `0.8.14`
